test/test_misc_utils.py

```diff
@@ -3727,3 +3727,35 @@ def test_create_readonly_object():
     assert a.ghi == 456
     assert a.jk == "xyzzy"
     assert a.lmnop == {"greeting": "Hello, world!"}
+
+
+def test_to_number():
+    assert to_number("1234") == 1234
+    assert to_number("1,234,567") is None
+    assert to_number("27500") == 27500
+    assert to_number("789", allow_float=True) == 789.0
+    assert type(to_number("789", allow_float=True)) == int
+    assert to_number("27500", allow_commas=True) == 27500
+    assert to_number("1,234,567", allow_commas=True) == 1234567
+    assert to_number("1234.0567", allow_float=True) == 1234.0567
+    assert to_number("1K", allow_multiplier_suffix=True) == 1000
+    assert to_number("1Kb", allow_multiplier_suffix=True) == 1000
+    assert to_number("1kB", allow_multiplier_suffix=True) == 1000
+    assert to_number("2M", allow_multiplier_suffix=True) == 2000000
+    assert to_number("2Mb", allow_multiplier_suffix=True) == 2000000
+    assert to_number("2MB", allow_multiplier_suffix=True) == 2000000
+    assert to_number("3G", allow_multiplier_suffix=True) == 3000000000
+    assert to_number("3Gb", allow_multiplier_suffix=True) == 3000000000
+    assert to_number("3GB", allow_multiplier_suffix=True) == 3000000000
+    assert to_number("4T", allow_multiplier_suffix=True) == 4000000000000
+    assert to_number("4Tb", allow_multiplier_suffix=True) == 4000000000000
+    assert to_number("4TB", allow_multiplier_suffix=True) == 4000000000000
+    assert to_number("1,234,567K", allow_commas=True) is None
+    assert to_number("1,234,567K", allow_commas=True, allow_multiplier_suffix=True) == 1234567000
+    assert to_number("-1,234,567K", allow_commas=True, allow_multiplier_suffix=True) == -1234567000
+    assert to_number(4321) == 4321
+    assert to_number(4321.1234) == 4321.1234
```

Comment:
What is the point of `allow_float` if the default behavior is to allow it? You also may want to be able to round to a specific digit. Does this test actually pass as written?

Generally I'm a little confused by `allow_float` - in my mind if passed it should always return a float, even if it is an integer, though this isn't what happens

Reply:
I reworked this, and tightened up tests.

Reply:
I reworked this, and tightened up tests.